executor/analyze.go

```diff
@@ -234,17 +234,21 @@ func analyzeIndexPushdown(idxExec *AnalyzeIndexExec) analyzeResult {
 		return analyzeResult{Err: err, job: idxExec.job}
 	}
 	result := analyzeResult{
-		TableID: idxExec.tableID,
-		Hist:    []*statistics.Histogram{hist},
-		Cms:     []*statistics.CMSketch{cms},
-		TopNs:   []*statistics.TopN{topN},
-		IsIndex: 1,
-		job:     idxExec.job,
+		TableID:  idxExec.tableID,
+		Hist:     []*statistics.Histogram{hist},
+		Cms:      []*statistics.CMSketch{cms},
+		TopNs:    []*statistics.TopN{topN},
+		IsIndex:  1,
+		job:      idxExec.job,
+		StatsVer: idxExec.ctx.GetSessionVars().AnalyzeVersion,
```

Comment:
Is this case possible: the user modified the variable when analyzing, which makes the stats version in `AnalyzePB` and the stats version in `analyzeResult` differ.
If so, we'd better save the stats version in the executor at first.